src/Microsoft.AspNetCore.Authentication.Core/AuthenticationService.cs

```diff
@@ -66,7 +66,7 @@ public AuthenticationService(IAuthenticationSchemeProvider schemes, IAuthenticat
             }
 
             var result = await handler.AuthenticateAsync();
-            if (result.Succeeded)
+            if (result?.Succeeded)
```

Comment:
I bet you $100 it will never compile (`result?.Succeeded` is a `bool?`).

You probably want to use `result != null && result.Succeeded` here.